src/js/actions/tools.js

```diff
@@ -418,7 +418,7 @@ define(function (require, exports) {
     selectTool.reads = [];
     selectTool.writes = [locks.JS_TOOL, locks.PS_TOOL];
     selectTool.transfers = [resetBorderPolicies, policy.removePointerPolicies, installShapeDefaults,
-        policy.removeKeyboardPolicies, policy.addPointerPolicies, policy.addKeyboardPolicies,
+        policy.removeKeyboardPolicies, policy.addPointerPolicies, policy.addKeyboardPolicies, policy.setMode,
```

Comment:
Ah, yes. The problem is that the tool select handlers aren't actually actions. They should be, and we should transfer to them. But, that's a problem for a different day. Thanks for the explanation.